# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmSignalHandlers.cpp
# Commit: b620533c6d63
# Full Hash: b620533c6d633ef50c5e5cb3ecd0f398206eb271
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-17 21:52:06
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r f7ec4a920540 -r b620533c6d63 js/src/wasm/WasmSignalHandlers.cpp
--- a/js/src/wasm/WasmSignalHandlers.cpp	Wed Jun 17 11:41:13 2020 +0000
+++ b/js/src/wasm/WasmSignalHandlers.cpp	Wed Jun 17 12:04:54 2020 +0000
@@ -707,7 +707,7 @@
   // due to this trap occurring in the indirect call prologue, while fp points
   // to the caller's Frame which can be in a different Module. In any case,
   // though, the containing JSContext is the same.
-  Instance* instance = ((Frame*)ContextToFP(context))->tls->instance;
+  Instance* instance = ((Frame*)ContextToFP(context))->instance();
   MOZ_RELEASE_ASSERT(&instance->code() == &segment.code() ||
                      trap == Trap::IndirectCallBadSig);
 
@@ -1167,7 +1167,7 @@
     return false;
   }
 
-  Instance& instance = *reinterpret_cast<Frame*>(regs.fp)->tls->instance;
+  Instance& instance = *Frame::fromUntaggedWasmExitFP(regs.fp)->instance();
   MOZ_ASSERT(&instance.code() == &segment.code());
 
   if (!instance.memoryAccessInGuardRegion((uint8_t*)addr, numBytes)) {